### PR TITLE
Add black. Update requirements. Fix pylint and isort.

### PR DESCRIPTION
**Pull request by tuchfarber**
Note this does *not* add black to CI. Not until we do the reformat.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
 
-.PHONY: help clean production-requirements js-requirements all-requirements requirements isort isort_check pycodestyle \
-        quality quality-js test-react tests js-tests static static.dev static.watch migrate up up-dev up-test \
-        exec-validate-translations exec-check_translations_up_to_date exec-check_keywords exec-pii_check exec-clean \
-        exec-requirements exec-static exec-quality exec-tests exec-accept exec-validate exec-coverage html_coverage \
-        shell tail stop down accept extract_translations dummy_translations compile_translations fake_translations \
-        pull_translations push_translations detect_changed_source_translations validate_translations \
-        check_translations_up_to_date piptools upgrade check_keywords pii_check
+.PHONY: help clean production-requirements js-requirements all-requirements requirements format_check format isort \
+        isort_check quality quality-js test-react tests js-tests static static.dev static.watch migrate \
```
Comment (crice100) [inline Makefile:6]: This formatting looks odd (why are the first 2 lines indented differently?), but sure why not

File: Makefile [hunk 2]
```diff
@@ -42,20 +42,22 @@ requirements: piptools ## Install requirements for local development
...
 isort_check: ## Check that isort has been run
 	isort --check-only -rc acceptance_tests/ credentials/
 
-pycodestyle: ## Run pycodestyle
-	pycodestyle acceptance_tests credentials *.py
-
-quality: ## Run linters
-	make isort_check
-	make pycodestyle
+quality: isort_check quality-js ## Run linters
```
Comment (crice100) [inline Makefile:57]: Does this need a _make_ in front of it?

File: pylintrc [hunk 3]
```diff
@@ -453,4 +482,4 @@ int-import-graph =
 [EXCEPTIONS]
 overgeneral-exceptions = Exception
 
-# d88146d61ffe6cf53d56f3284a5e2fc78667ba56
+# 3b6ad6681ffd9487a55a8ebd882eead12d707e81
```
Comment (crice100) [inline pylintrc:485]: Just curious--what is this?

File: Makefile (hunk 1 above)
Comment (tuchfarber) [reply Makefile]: Ugh. This is a tabs/spaces issue, so it looked fine locally. Should be fixed now.

File: Makefile (hunk 2 above)
Comment (tuchfarber) [reply Makefile:57]: Nope. It only needs `make` if it's a step in the command. This instead says that `isort_check` and `quality-js` are prerequisites, so those commands should be ran first.

File: pylintrc (hunk 3 above)
Comment (tuchfarber) [reply pylintrc:485]: No idea. Guessing a hash of the the rules? I needed to add overrides, so I had to edit `pylintrc_tweaks`, then run an `edx_lint write pylintrc` and this is what came out. :shrug:

File: Makefile [hunk 4]
```diff
@@ -2,13 +2,13 @@
...
-        shell tail stop down accept extract_translations dummy_translations compile_translations fake_translations \
-        pull_translations push_translations detect_changed_source_translations validate_translations \
-        check_translations_up_to_date piptools upgrade check_keywords pii_check
+.PHONY: help clean production-requirements js-requirements all-requirements requirements format_check format isort \
+	isort_check quality quality-js test-react tests js-tests static static.dev static.watch migrate \
+	up up-dev up-test exec-validate-translations exec-check_translations_up_to_date exec-check_keywords \
+	exec-pii_check exec-clean exec-requirements exec-static exec-quality exec-tests exec-accept exec-validate \
+	exec-coverage html_coverage shell tail stop down accept extract_translations dummy_translations \
+	compile_translations fake_translations pull_translations push_translations detect_changed_source_translations \
+	validate_translations check_translations_up_to_date piptools upgrade check_keywords pii_check
```
Comment (crice100) [inline Makefile:11]: Nit: I think this is missing _quality_fix_

**Review by crice100 — APPROVED** [review]

**Pull request merged**